node/src/service.rs

```diff
@@ -251,11 +255,7 @@ pub async fn new_full<Network: sc_network::NetworkBackend<Block, <Block as Block
 		let shared_voter_state = shared_voter_state.clone();
 		let shared_authority_set = grandpa_link.shared_authority_set().clone();
 		let justification_stream = grandpa_link.justification_stream();
-		let main_chain_follower_data_sources =
-			crate::main_chain_follower::create_cached_main_chain_follower_data_sources(
-				mc_follower_metrics_opt,
-			)
-			.await?;
+		let main_chain_follower_data_sources = data_sources.clone();
```

Comment:
I think this should have been fixed a long time ago, when "safe caching" was implemented.